csharp/ql/src/semmle/code/csharp/ir/implementation/raw/internal/TranslatedCall.qll

```diff
@@ -5,287 +5,67 @@ private import InstructionTag
-  }
-
-  /**
-   * Gets the result type of the call.
-   */
-  abstract Type getCallResultType();
-
-  /**
-   * Holds if the call has a `this` argument.
-   */
-  predicate hasQualifier() {
-    exists(this.getQualifier())
-  }
-
-  /**
-   * Gets the `TranslatedExpr` for the indirect target of the call, if any.
-   */
-  TranslatedExpr getCallTarget() { none() }
-
-  /**
-   * Gets the first instruction of the sequence to evaluate the call target.
-   * By default, this is just the first instruction of `getCallTarget()`, but
-   * it can be overridden by a subclass for cases where there is a call target
-   * that is not computed from an expression (e.g. a direct call).
-   */
-  Instruction getFirstCallTargetInstruction() {
-    result = this.getCallTarget().getFirstInstruction()
-  }
-
-  /**
-   * Gets the instruction whose result value is the target of the call. By
-   * default, this is just the result of `getCallTarget()`, but it can be
-   * overridden by a subclass for cases where there is a call target that is not
-   * computed from an expression (e.g. a direct call).
-   */
-  Instruction getCallTargetResult() {
-    result = this.getCallTarget().getResult()
-  }
-
-  /**
-   * Gets the `TranslatedExpr` for the qualifier of the call (i.e. the value
-   * that is passed as the `this` argument.
-   */
-  abstract TranslatedExpr getQualifier();
-
-  /**
-   * Gets the instruction whose result value is the `this` argument of the call.
-   * By default, this is just the result of `getQualifier()`, but it can be
-   * overridden by a subclass for cases where there is a `this` argument that is
-   * not computed from a child expression (e.g. a constructor call).
-   */
-  Instruction getQualifierResult() {
-    result = this.getQualifier().getResult()
-  }
-
-  /**
-   * Gets the argument with the specified `index`. Does not include the `this`
-   * argument.
-   */
-  abstract TranslatedExpr getArgument(int index);
-
-  /**
-   * If there are any arguments, gets the first instruction of the first
-   * argument. Otherwise, returns the call instruction.
-   */
-  final Instruction getFirstArgumentOrCallInstruction() {
-    if this.hasArguments() then
-      result = this.getArgument(0).getFirstInstruction()
-    else
-      result = this.getInstruction(CallTag())
-  }
-
-  /**
-   * Holds if the call has any arguments, not counting the `this` argument.
-   */
-  abstract predicate hasArguments();
-
-  // TODO: Fix side effects
-  predicate hasReadSideEffect() { any() }
-
-  predicate hasWriteSideEffect() { any() }
-
-  private predicate hasSideEffect() { hasReadSideEffect() or hasWriteSideEffect() }
-
-  override Instruction getPrimaryInstructionForSideEffect(InstructionTag tag) {
-      this.hasSideEffect() and
-      tag = CallSideEffectTag() and
-      result = this.getResult()
+  
+  override Instruction getUnmodeledDefinitionInstruction() {
+    result = this.getEnclosingFunction().getUnmodeledDefinitionInstruction()
```

Comment:
Might it be more convenient to have a `override ... getTranslatedFunction()` instead? Then use `getTranslatedFunction().getUnmodeledDefinitionInstruction()` etc.

Reply:
There already is a predicate `getTranslatedFunction/1`, so this could be a bit confusing.